fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBEnvironment.java

```diff
@@ -465,100 +457,30 @@ public ReplicatedEnvironment getReplicatedEnvironment() {
-                    iter.remove();
-                    continue;
-                }
-
-                if (openedDb.getDb().getDatabaseName().equals(dbName)) {
-                    return openedDb;
-                }
-            }
-
-            // open the specified database.
-            // the first parameter null means auto-commit
-            try {
-                db = new CloseSafeDatabase(replicatedEnvironment.openDatabase(null, dbName, dbConfig));
-                openedDatabases.add(db);
-                LOG.info("successfully open new db {}", db);
-            } catch (Exception e) {
-                LOG.warn("catch an exception when open database {}", dbName, e);
-            }
-        } finally {
-            lock.writeLock().unlock();
-        }
-        return db;
+        return new CloseSafeDatabase(replicatedEnvironment.openDatabase(null, dbName, dbConfig));
     }
 
-    // close and remove the database whose name is dbName
+    /**
+     * Remove the database whose name is dbName
+     * We should make sure no database conflict from upper level
+     **/
     public void removeDatabase(String dbName) {
-        lock.writeLock().lock();
+        if (closing) {
+            return;
+        }
+
         try {
-            if (closing) {
-                return;
-            }
-            String targetDbName = null;
-            int index = 0;
-            for (CloseSafeDatabase db : openedDatabases) {
-                String name = db.getDb().getDatabaseName();
-                if (dbName.equals(name)) {
-                    db.close();
-                    LOG.info("database {} has been closed", name);
-                    targetDbName = name;
-                    break;
-                }
-                index++;
-            }
-            if (targetDbName != null) {
-                LOG.info("begin to remove database {} from openedDatabases", targetDbName);
-                openedDatabases.remove(index);
-            }
-            try {
-                LOG.info("begin to remove database {} from replicatedEnviroment", dbName);
-                // the first parameter null means auto-commit
-                replicatedEnvironment.removeDatabase(null, dbName);
-            } catch (DatabaseNotFoundException e) {
-                LOG.warn("catch an exception when remove db:{}, this db does not exist", dbName, e);
-            }
-        } finally {
-            lock.writeLock().unlock();
+            LOG.info("begin to remove database {} from replicatedEnviroment", dbName);
+            // the first parameter null means auto-commit
+            replicatedEnvironment.removeDatabase(null, dbName);
+            LOG.info("remove database {} from replicatedEnviroment successfully", dbName);
+        } catch (DatabaseNotFoundException e) {
+            LOG.warn("catch an exception when remove db:{}, this db does not exist", dbName, e);
```

Comment:
LOG.warn("Exception: {} does not exist", dbName, e);

Reply:
Done